### ISSUE TITLE
Questions about unit testing

### ISSUE THREAD

**Issue by weigui** [dhuang12138/CS2820#14]
Now our project has been divided into two part:
The main production part (package production), Which is for THE DEMO DAY.
and the unitTesting (package production_unitTesting) part.

The main production part was almost entirely written by our master role contributor Di Huang, because for the Demo Day we need a PERFECTLY RUNNABLE system with a friendly visualizer.

And the unitTesting part is for YOUR project evaluation grades: make your own subsystem and testing class with a Main method here. 
PLEASE DO NOT USE any methods or fields from the main production package. If you need any help writing your own class and testing method( like my Robot need the getRoute() method in Floor class) , please let other people know.

**Comment by josephLeiferman** [comment]
Easy there Wei Gui, lets not minimize the fact we all have been working hard on the project not just the master role we all have contributed a lot. The work that Di Huang did was great, but so was everyone's else. Given more time and better communication I believe everyone's code that they wrote would have been used.

**Comment by di-huang** [comment]
weigui, you said you will add some magical functions in Visualizer and you look like very active and motivated about that, so you should be the one who demonstrate your magic visualization xD

**Comment by weigui** [comment]
@josephLeiferman 
Err... we may have some kind of misunderstanding here... I'm just explaining what we've already got in our project.
 I mean, in the unitTesting package we shouldn't implement the production package because that would mess things up: they are two different things!

In the unitTesting package we should focus on our own class: fields,methods,comments. Eliminate those errors and warnings in your class, and then write a good testing main method for it.——That's how they grade your own work. 

That would be awesome if everybody is working hard to do something in the main production part, which is for our demo day. We need to show that our group can complete this project and make the whole warehouse system work, then your individual subsystem and testing method.

**Comment by josephLeiferman** [comment]
@weigui Oh no I understand you, I didn't mean to cause any conflict. I understand we want to keep the production package working so that we have something to demo. I just wanted to make sure no one felt slighted, if some of their code didn't make it into demo even if they had very similar code already complete.

**Comment by josephLeiferman** [comment]
We just had a cohesion problem, where we were all working in a 'bubble', and not communicating so work overlapped or didn't work together properly that is why some people's code had to be re-written.

**Comment by weigui** [comment]
:i Anyway, If anybody need help for unitTesting package, please feel free to comment here

**Comment by NickEBarnes** [comment]
Just trying to get it working, I really don't know why it isn't getting to the main in production

________________________________

**Comment by weigui** [comment]
@NickEBarnes 
You mean, Eclipse won't run your main method?

**Comment by josephLeiferman** [comment]
What is your error message?

**Comment by NickEBarnes** [comment]
Well I guess the main error right now is that Item and other classes have been changed and when I try to run it it just blows up. There are a lot of classes that seem to be different than what they were. Are you guys able to run the unitTesting?

________________________________

**Comment by NickEBarnes** [comment]
It isn't my main method, it is the production main method that it had been using

________________________________

**Comment by weigui** [comment]
@NickEBarnes 
Don't use the production main method, because the unitTesting is not a runnable warehouse_system
You need to write your own testing main method which will tell the TAs that your subsystem can handle its mission

**Comment by NickEBarnes** [comment]
It just kinda all got messed up, I hadn't committed my stuff which I admit I can only blame myself, but then I committed/pulled and then it was all gone so yeah. I can do my own main method that's fine.

________________________________

**Comment by josephLeiferman** [comment]
I agree with weigui the best solution is probably create your own class to test the work you did like beltTest.java in production_unitTesting .

**Comment by di-huang** [comment]
@NickEBarnes It seems like you have to add what you need (like method, variable...) to MockObject which you're gonna call, so that's what MockObject does. For example, your role is Belt and you may use some methods of other role (like Floor), then you can just add what you need in MockFloor. What's more, you also have to build a TestBelt class to finish your own unit test.

**Comment by weigui** [comment]
That's right, your mockXXX class needs to pretend that other subsystem are working properly ( even though they are not)
That's why you need your own main method.

**Comment by di-huang** [comment]
Every role has its own interface (except Master) which provides some basic method signature, and its MockObject will implement it, but if you find there's still something else you need from that MockObject, you can just add it to that MockObject.

**Comment by NickEBarnes** [comment]
Right right, I understand what you are saying. I just had everything going through Production earlier, I'll work on it. Props on the demo project, looks nice

________________________________

**Comment by di-huang** [comment]
Also, you can just ignore the errors that occur in other MockObject, because he has responsiblity to fix it on his own, and you should still be able to call the methods you need from it (if doesn't exist, just add it). However, you have to assume your MockObject actually does something you expect, so you need to report what it does at each tick (you need to assume you already receive the central tick from Master, so you can add a tick variable for unit test and "tick++" when tick(clk) method is called)by printing on console(method-stub).

**Comment by ghost** [comment]
Thank you for making your decisions clearer Di Huang and Wei Gui, I was confused yesterday of the status of the project.

**Comment by charlesc3929** [comment]
Still having trouble getting the project onto my computer. Hopefully one of you guys knows Git better than I do, but I have a f
